cpp/src/reductions/scan.cu

```diff
@@ -33,37 +33,57 @@ struct ScanDispatcher {
     return std::is_arithmetic<T>::value;
   }
 
+  // @brief creates new column of same size, and copies null mask of input column view.
+  std::unique_ptr<column> allocate_like_copied_nullmask(const column_view& input,
+      rmm::mr::device_memory_resource* mr, cudaStream_t stream)
+  {
+    return make_numeric_column(
+        input.type(), input.size(), 
+        copy_bitmask(input, stream, mr), //copy bit mask
+        input.null_count(), stream, mr);
+  } 
+
   template <typename T, typename InputIterator>
-  void exclusive_scan(const InputIterator input, T* output, size_t size,
+  auto exclusive_scan(const InputIterator input, const column_view& input_view,
```

Comment:
It's a bit weird to take an input iterator and an input column_view. It looks like the dispatcher functor just creates the iterator from the view. So maybe just do that inside this function and pass only the view?

Reply:
I could. I felt i'm doing too many operations at this leaf level function. so I segregated iterator creation outside this function, and output column creation inside this function. I felt that it's weird to pass input iterator and input_view. input_view is passed to create output column with copied nullmask of. 

{Iterator creation, Output column/scalar creation, Actual computation}
This will become repeated pattern at many places. Some suggestions on handling this in future  would be good. Probably add your suggestions to transition guide as well.

Reply:
and also, since iterator type is different for nullable and non-nullable column, cub:: scan call code should be repeated twice inside `if(col.nullable())` and `else`.